Data/Aeson/Types/Class.hs

```diff
@@ -272,3 +320,95 @@ typeMismatch expected actual =
              Number _ -> "Number"
              Bool _   -> "Boolean"
              Null     -> "Null"
+
+-------------------------------------------------------------------------------
+-- Lifings of FromJSON and ToJSON to unary and binary type constructors
+-------------------------------------------------------------------------------
+
+-- | Lifting of the 'FromJSON' class to unary type constructors.
+class FromJSON1 f where
+    liftParseJSON :: (Value -> Parser a) -> (Value -> Parser [a]) -> Value -> Parser (f a)
+    liftParseJSONList :: (Value -> Parser a) -> (Value -> Parser [a]) -> Value -> Parser [f a]
+    liftParseJSONList f g v = case v of
+        Array vals -> fmap V.toList (V.mapM (liftParseJSON f g) vals)
+        _ -> typeMismatch "[a]" v
```

Comment:
We should really put this functionality into a combinator of type `Parser a -> Parser [a]` like we did for `listValue`.